seyren-aws-manager/src/main/java/com/seyren/awsmanager/cache/AWSInstanceDetailsCache.java

```
@@ -0,0 +1,61 @@
+package com.seyren.awsmanager.cache;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.seyren.awsmanager.entity.AWSInstanceDetail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.jar.Pack200;
+
+/**
+ * Created by akharbanda on 04/09/17.
+ */
+public class AWSInstanceDetailsCache
+{
+    private final LoadingCache<String,AWSInstanceDetail> awsInstanceDetailLoadingCache ;
+    private final CacheLoader<String,AWSInstanceDetail> cacheLoader;
+    private final CacheBuilder cacheBuilder ;
+    private static final Logger LOGGER = LoggerFactory.getLogger(AWSInstanceDetailsCache.class);
+
+    public AWSInstanceDetailsCache(CacheLoader cacheLoader , Long maxSize , Long expiryTimeoutinMillis)
+    {
+        this.cacheLoader = cacheLoader;
+        cacheBuilder = CacheBuilder.newBuilder();
+        if(maxSize!=null)
+        {
+            cacheBuilder.maximumSize(maxSize);
+        }
+
+        if(expiryTimeoutinMillis!=null)
+        {
+            cacheBuilder.expireAfterWrite(expiryTimeoutinMillis, TimeUnit.MILLISECONDS);
+        }
+
+        awsInstanceDetailLoadingCache = cacheBuilder.build(cacheLoader);
+    }
+
+    public AWSInstanceDetail getAWSInstanceDetails(String key)
+    {
+        AWSInstanceDetail awsInstanceDetail = null;
+
+        try{
+            awsInstanceDetail = awsInstanceDetailLoadingCache.get(key);
+        }
+        catch (ExecutionException ee)
+        {
+        }
```

you may want to just throw new runtime exception here.

catch (ExecutionException ee) {
    throw new RuntimeException(ee);
}

---

We do not want to rethrow the exception . Let me explain why this is required here . The Guava autoloading cache invokes the cache loader every time it does not find a value in cache. However , I do not want to invoke the cache loader for each instance but make one AWS call to get details for all instances not in cache . 
The way to achieve this is to throw an exception from CacheLoader load method (class AWSManager.java line 76) and catch it in here .

---

ahh ok thanks.